### PR TITLE
adds case insensitive lookup for AssociaationCollection::type()

### LINKED ISSUE #9056: Can AssociationCollection::type() throw an exception for unknown types?
This is a (multiple allowed):
* [ ] bug
* [x] enhancement
* [ ] feature-discussion (RFC)

* CakePHP Version: 3.2.10

### What you did

Tried to get all the belongsTo associations

```
        $this->belongsTo('Nodes');
        $this->belongsTo('Jobs');
        debug($this->associations()->type('belongsTo'));
```

### Expected Behavior

See a list of belongsTo assocation.

### Actual Behavior

Association type is case sensitive and I should have used "BelongsTo", but instead it returns an empty string. Which was very confusing and took me a while to spot my mistake.

What do you think about `type()` throwing an exception if the type is not supported, and how would the method know?

### PR DESCRIPTION
**Pull request by thinkingmedia**
Closes #9056

### CODE REVIEW

File: tests/TestCase/ORM/AssociationCollectionTest.php [hunk 1]
```diff
@@ -144,19 +144,35 @@ public function testKeys()
...
-        $this->assertSame(
-            [$belongsTo, $belongsToMany],
-            $this->associations->type(['BelongsTo', 'BelongsToMany'])
-        );
+        foreach ($belongsToCases as $belongsToType) {
+            $this->assertSame(
+                [$belongsTo],
+                $this->associations->type($belongsToType)
+            );
+            foreach ($belongsToManyCases as $belongsToManyType) {
```
Comment (markstory) [inline tests/TestCase/ORM/AssociationCollectionTest.php]: Instead of nested loops, could data providers be used?

**Comment by thinkingmedia** [comment]
done.

**Comment by ADmad** [comment]
@thinkingmedia Can you please squash the commits into one?

**Comment by thinkingmedia** [comment]
@ADmad Don't you have the option when accepting the PR?

https://help.github.com/articles/about-pull-request-merge-squashing/

**Comment by thinkingmedia** [comment]
@ADmad it has to be enabled in the project settings. Maybe it's turned off.

**Pull request merged**